src/function_benchmark.rs

```diff
@@ -0,0 +1,145 @@
+use colored::Colorize;
+use std::{fmt, time::Duration};
+
+const RUNTIME_THRESHOLD: Duration = Duration::from_millis(5);
+
+pub struct FunctionBenchmark {
```

Comment:
Nit: Maybe name `BenchmarkPrinter`? Not really doing any benchmarking but more of a `Printer` type concept. It takes in some numbers and nicely prints them.

Reply:
Good idea, this name is an artefact from a previous implementation that I refactored

Reply:
RunStatistics sounds cool, I'll use this for now